src/StorageWriter/ClaimController.ts

```diff
@@ -1,51 +1,112 @@
 import { Claim } from '@po.et/poet-js'
 import { inject, injectable } from 'inversify'
-import { Collection, Db } from 'mongodb'
 import * as Pino from 'pino'
+import { pipeP, lensProp, lensPath, set, view } from 'ramda'
 
 import { childWithFileName } from 'Helpers/Logging'
-import { Exchange } from 'Messaging/Messages'
-import { Messaging } from 'Messaging/Messaging'
 
+import { Database } from './Database'
+import { uploadClaimError, updateClaimHashError } from './Errors'
 import { IPFS } from './IPFS'
 
+enum LogTypes {
+  'info' = 'info',
+  'trace' = 'trace',
+  'error' = 'error',
+}
+
+interface StoreNextClaimData {
+  claim: Claim
+  ipfsFileHash?: string
+}
+
+const L = {
+  claim: lensProp('claim'),
+  ipfsFileHash: lensProp('ipfsFileHash'),
+  claimId: lensPath(['claim', 'id']),
+}
+
 @injectable()
 export class ClaimController {
   private readonly logger: Pino.Logger
-  private readonly db: Db
-  private readonly collection: Collection
-  private readonly messaging: Messaging
+  private readonly db: Database
   private readonly ipfs: IPFS
 
-  constructor(
-    @inject('Logger') logger: Pino.Logger,
-    @inject('DB') db: Db,
-    @inject('Messaging') messaging: Messaging,
-    @inject('IPFS') ipfs: IPFS
-  ) {
+  constructor(@inject('Logger') logger: Pino.Logger, @inject('Database') db: Database, @inject('IPFS') ipfs: IPFS) {
     this.logger = childWithFileName(logger, __filename)
     this.db = db
-    this.collection = this.db.collection('storageWriter')
-    this.messaging = messaging
     this.ipfs = ipfs
   }
 
-  async create(claim: Claim): Promise<void> {
-    const logger = this.logger.child({ method: 'create' })
+  private readonly log = (level: LogTypes) => (message: string) => async (value: any) => {
```

Comment:
Move to LogHelper? Might want to reuse this in other places

Reply:
Sounds good to me. You ok with me doing this in a separate PR?

Reply:
Yep